lib/matplotlib/colorbar.py

```diff
@@ -490,12 +490,13 @@ def update_normal(self, mappable):
         changes values of *vmin*, *vmax* or *cmap* then the old formatter
         and locator will be preserved.
         """
-        _log.debug('colorbar update normal %r %r', mappable.norm, self.norm)
-        self.mappable = mappable
-        self.set_alpha(mappable.get_alpha())
-        self.cmap = mappable.cmap
-        if mappable.norm != self.norm:
-            self.norm = mappable.norm
+        if mappable:
+            self.mappable = mappable
```

Comment:
`self.mappable` should never be None, as a `cm.ScalarMappable` is created if `mappable=None` on the creation of the colorbar. 

These code lines remain because they are in the existing codebase, and I don't want to change more than necessary.

Note that the colorbar currently connects to an `ColorizingArtist` (or a `ScalarMappable`). It would be more logically consistent to connect it to a `Colorizer` object, and this should be changed in a later PR.

Also, there is a difference in how `ScalarMappable` and `ColorizingArtist` (and `Colorizer`) emits changed signals. 
I added the following note:
```py
def update_normal(self, mappable=None):
    if mappable:
        # The mappable keyword argument exists because 
        # cm.ScalarMappable.changed() emits self.callbacks.process('changed', self)
        # in contrast, ColorizingArtist (and Colorizer) does not use this keyword.
        # [colorizer.ColorizingArtist.changed() emits self.callbacks.process('changed')]
        # Also, there is no test where self.mappable == mappable is not True and possibly no use case. 
        # Therefore, the mappable keyword can be depreciated if cm.ScalarMappable is removed.
        self.mappable = mappable
  ```